pkg/githubclient/client_repository.go

```diff
@@ -52,3 +53,175 @@ func (c *Client) ListRepositories(ctx context.Context, owner string) ([]Reposito
 
 	return repos, nil
 }
+
+func (c *Client) GetRepository(ctx context.Context, owner, repo string) (*github.Repository, error) {
+	c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("get repository %s/%s", owner, repo))
+
+	underlyingClient := c.getUnderlyingClient(ctx)
+
+	repository, response, err := underlyingClient.Repositories.Get(ctx, owner, repo)
+	if err != nil {
+		if response != nil && response.Response != nil && response.Response.StatusCode == http.StatusNotFound {
+			return nil, microerror.Mask(notFoundError)
+		}
+		return nil, microerror.Mask(err)
+	}
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("got repository %s", *repository.FullName))
+
+	return repository, nil
+}
+
+func (c *Client) SetRepositorySettings(ctx context.Context, repository, repositorySettings *github.Repository) (*github.Repository, error) {
+	owner := *repository.Owner.Login
+	repo := *repository.Name
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting repository %s/%s settings", owner, repo))
+
+	// Features
+	repository.HasWiki = repositorySettings.HasWiki
+	repository.HasIssues = repositorySettings.HasIssues
+	repository.HasProjects = repositorySettings.HasProjects
+	repository.Archived = repositorySettings.Archived
+
+	// Merge settings
+	repository.AllowMergeCommit = repositorySettings.AllowMergeCommit
+	repository.AllowSquashMerge = repositorySettings.AllowSquashMerge
+	repository.AllowRebaseMerge = repositorySettings.AllowRebaseMerge
+
+	// Pull Requests
+	repository.AllowUpdateBranch = repositorySettings.AllowUpdateBranch
+	repository.AllowAutoMerge = repositorySettings.AllowAutoMerge
+	repository.DeleteBranchOnMerge = repositorySettings.DeleteBranchOnMerge
+
+	// This is required since Github does not allow overrides for flags specified
+	// at organization level.
+	// Otherwise you will run into the following error:
+	// HTTP 422 This organization does not allow private repository forking
+	repository.AllowForking = nil
+
+	underlyingClient := c.getUnderlyingClient(ctx)
+	repository, _, err := underlyingClient.Repositories.Edit(ctx, *repository.Owner.Login, *repository.Name, repository)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("set repository %s/%s settings", owner, repo))
+
+	return repository, nil
+}
+
+func (c *Client) SetRepositoryPermissions(ctx context.Context, repository *github.Repository, permissions map[string]string) error {
+	org := *repository.Organization.Login
+	owner := *repository.Owner.Login
+	repo := *repository.Name
+
+	//data, err := json.MarshalIndent(permissions, "", "  ")
```

Comment:
Missing removal?

Reply:
Good point, I'll rework that while re-doing the log output.